sdk/metric/aggregator/gauge/gauge.go

```diff
@@ -125,3 +123,34 @@ func (g *Aggregator) updateMonotonic(number core.Number, desc *export.Descriptor
 		}
 	}
 }
+
+func (g *Aggregator) Merge(oa export.MetricAggregator, desc *export.Descriptor) {
+	o, _ := oa.(*Aggregator)
+	if o == nil {
+		// TODO warn
+		return
+	}
+
+	ggd := (*gaugeData)(atomic.LoadPointer(&g.checkpoint))
+	ogd := (*gaugeData)(atomic.LoadPointer(&o.checkpoint))
+
+	if desc.Alternate() {
+		// Monotonic: use the greater value
+		cmp := ggd.value.CompareNumber(desc.NumberKind(), ogd.value)
+
+		if cmp > 0 {
+			return
+		}
+
+		if cmp < 0 {
+			g.checkpoint = unsafe.Pointer(ogd)
+			return
+		}
```

Comment:
Does it even make sense to merge two monotonic gauges? Should it be sum?

Reply:
Merge will be used when the SDK has (because it tolerates race conditions / is lock-free) duplicate records. The correct action for the gauge aggregator is to take the last value.  For sum behavior, you'd use the a counter aggregator. Note that any aggregator can be attached to any metric -- you could sum a gauge instrument by attaching a counter aggregator.

Reply:
Is Merge strictly used for combining two or more data points that resulted due to race condition in the same Collect cycle?

Reply:
It is also useful to "group by", when reducing dimensionality. I'll be sending a stdout exporter soon that depends on this code.

Reply:
No real life example comes to my mind for "group by" WRT monotonic gauges.  
Anyway, LGTM otherwise.

Reply:
Because of LabelSets, the user can supply as many labels as they want, but most of today's exporters export over fixed dimensions. As the user could have supplied extra labels, we need to reduce dimensionality. Perhaps it would make more sense if I renamed this aggregator "lastvalue". It's the default for aggregating gauges. 

Perhaps you are thinking that "counter" is a more suitable default aggregation for monotonic gauges? That I might agree to, but the semantics of a gauge instrument are "Set". I don't see why we should add them by default. (I think monotonic gauges are a farce intended for a legacy system, but someone apparently wants them.)